### ISSUE TITLE
Making custom docker container publicly accessible through marathon-lb

### ISSUE THREAD

**Issue by marceldempers** [mesosphere/marathon-lb#375]
Hi,

I am using Azure Container Service and have a healthy MESOS cluster.
We have written a custom application in nodejs that we build and push to custom registry.
I can successfully host this on my local docker setup and browse to its HTTP home page.

I am using marathon to deploy this application to my cluster.
I have added marathon-lb and can host multiple sample applications successfully, but for the life of me I cannot get this custom application accessed via the marathon-lb (from public URL).

Everything in my marathon template is fairly standard, and as far as I am aware, all I need to do is use VHOST as per below

```
"labels": {
  "HAPROXY_GROUP": "external",
  "HAPROXY_0_VHOST": "mydomain.com",
  "HAPROXY_0_MODE": "http"
}
```

mydomain.com is "host filed" for testing as I don't have a DNS at this stage so I point that at my agents public ip address.

My port mappings are also straight forward as I leave host port to be randomized by the system and container port I run on 3333.

```
"portMappings": [
   {
      "containerPort": 3333,
      "hostPort" : 0,
      "servicePort" : 10003,
      "protocol": "tcp"
   }
],
```

If I deploy this as above, it deploys successfully and I can see by the logs that my application is running.
I can SSH to the master, then to the running agent, then use containers IP and port shown in marathon and run:
curl http://containerIP:[hostport] and it works successfully. I can see my home page.

What does not work is when I am on master or any of the agents and use the marathon-lb DNS endpoint, it gives me a "503 Service Unavailable: No server is available to handle this request."

If I use my host file entry on my machine I can hit it from outside and still get the 503 as well.

So I understand my container is running as I can troubleshoot it as above, but I am missing some key piece of info about the marathon lb configuration that I just cannot get right.

I am able to deploy image "yeasy/simple-web" and "tutum/hello-world" sample apps exacly the same way and I can access it from outside using my host file entry through the marathon-lb.
Whats interesting if I deploy these sample apps, and try use another container port other than 80, like 3334 or 3335, i get the 503 as well.

Is there someone who can please shed some light on the matter and can guide me to get this working ?

Thanks a million,
Marcel

**Comment by marceldempers** [comment]
To give more clarity on two containers I have running, I have troubleshooted the container and can see I am  unable to perform CURL on the mesos endpoint for my custom application on its service endpoint, but it succeeds on a sample "simple web" from docker hub. See my visio diagram:
![image](https://cloud.githubusercontent.com/assets/11763725/20546633/ac472a24-b16a-11e6-911c-1544afe41d8e.png)

**Comment by brndnmtthws** [comment]
Can you fetch the haproxy config? `curl marathon-lb.marathon.mesos:9090/_haproxy_getconfig`.

Is the app inside the container binding to port 3333, or port 80? If it's 80, set `containerPort` to 80 rather than 3333.

**Comment by marceldempers** [comment]
@brndnmtthws 
Thanks for you response.
I have managed to resolve the issue so we can close this one.

For others out there who may run into this -
The problem occurred because the container application is running an express web server that was configured in a file to always listen on a specific hard coded port rather than 80. So by setting a port in Marathon, it had no impact hence the marathon-lb could not reach the application.

There was a environment variable required to override this to 80, but I had to do quite a bit of digging through our code to find that out
